src/main/java/javaslang/collection/Seq.java

```diff
@@ -198,6 +198,25 @@ default boolean containsSlice(java.lang.Iterable<? extends T> that) {
     }
 
     /**
+     * Tests whether every element of this sequence relates to the corresponding element of another sequence by satisfying a test predicate.
+     *
+     * @param that the other sequence
+     * @param predicate the test predicate, which relates elements from both sequences
+     * @return {@code true} if both sequences have the same length and {@code predicate(x, y)}
+     * is {@code true} for all corresponding elements {@code x} of this sequence and {@code y} of {@code that}, otherwise {@code false}.
+     */
+    default <U> boolean corresponds(java.lang.Iterable<U> that, Predicate<Tuple2<T, U>> predicate) {
```

Comment:
As already mentioned I would change the bounds:

```java
default <U> boolean corresponds(java.lang.Iterable<? extends U> that, Predicate<Tuple2<? super T, ? super U>> predicate)
```

Also I would move corresponds to `javaslang.Iterable`.

Use case: The following RedBlackTrees (and therefore SortedSets) are equal/corresponding:

```java
      -1       -1
      / \      / \
    -2   1   -2   0
        /          \
       0            1
```

because both Iterators return elements in the following order (tree in-order): `-2, -1, 0, 1`.